cancat/ccp/ccp_leader.py

```diff
@@ -0,0 +1,1572 @@
+        |   2   |  byte  |  Mode: start/stop/prepare data tranmission    |
+        +-------+--------+-----------------------------------------------+
+        |   3   |  byte  |  DAQ list number                              |
+        +-------+--------+-----------------------------------------------+
+        |   4   |  byte  |  Last ODT number                              |
+        +-------+--------+-----------------------------------------------+
+        |   5   |  byte  |  Event Channel No.                            |
+        +-------+--------+-----------------------------------------------+
+        |  6,7  |  word  |  Transmission rate prescaler                  |
+        +-------+--------+-----------------------------------------------+
+        '''
+
+        # this one is motorola instead of intel format, according to the spec
+        trans_rate_prescaler = struct.pack('>H', transmission_rate_prescaler)
+
+        data = utils._gen_byte(mode) + utils._gen_byte(daq_list_number) + \
+               utils._gen_byte(last_odt_num) + utils._gen_byte(event_chan_num) + \
+               trans_rate_prescaler
+
+        msg = self._constructCRO(CCP_START_STOP, counter, data)
+
+        return msg
+
+    def _start_stop_all_CRO(self, counter, start_or_stop):
+        '''
+        START_STOP_ALL CRO
+
+        data to be sent after CMD and CTR values:
+        +-------+--------+-----------------------------------------------+
+        |   2   |  byte  |  0x00 stops, 0x01 starts data transmission    |
+        +-------+--------+-----------------------------------------------+
+        |  3..7 |  bytes |  don't care                                   |
+        +-------+--------+-----------------------------------------------+
+        '''
+
+        data = utils._gen_byte(start_or_stop) + utils._gen_byte(DONT_CARE_VAL)*5
+
+        msg = self._constructCRO(CCP_START_STOP_ALL, counter, data)
+
+        return msg
+
+    '''
+    +---------------------------------------------------------------------+
+    |                                                                     |
+    |                 Data Transmission Object functions                  |
+    |                                                                     |
+    +---------------------------------------------------------------------+
+    '''
+
+    def _parse_DTO(self, CCP_message):
+        '''
+        Data Transmission Object (DTO): message sent from follower to leader
+
+        Can be one of three types:
+        * Command Return Message: follower sending message back in response to CRO from leader
+        * Event Message: reporting internal status change in order to invoke error recovery
+        * Data Acquisition Message: Packet ID must correspond to ODT (Object Descriptor Table)
+        '''
+
+        if len(CCP_message) != 8:
+            raise Exception("CCP message should have length 8")
+
+        if CCP_message[0] == b'\xFF':
```

Comment:
does this work?  in Py3, b'ABCDEF'[0] == 0x41  (an int, not bytes).
in order to make this a thing, you need to use [0:1] or 0xFF (not bytes).

can you confirm??